experiments/hash_mutants.py

```diff
@@ -0,0 +1,40 @@
+__all__ = ['mutation_to_uid']
+
+import argparse
+import hashlib
+
+from ruamel.yaml import YAML
+from ruamel.yaml.scalarstring import PreservedScalarString as PSS
+
+DESCRIPTION = """
+Adds unique identifiers to each mutant within a ground truth database file.
+""".strip()
+
+
+def mutation_to_uid(diff: str) -> str:
+    m = hashlib.sha512()
+    m.update(diff.encode())
+    uid = m.hexdigest()[:24]
+    return uid
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description=DESCRIPTION)
+    parser.add_argument('input', type=str,
+                        help="path to input ground truth database file.")
+    parser.add_argument('output', type=str,
+                        help="path to output ground truth database file.")
+    return parser.parse_args()
+
+
+if __name__ == '__main__':
+    args = get_args()
+    with open(args.input, 'r') as f:
+        db = YAML().load(f)
+
+    for entry in db['entries']:
+        entry['uid'] = mutation_to_uid(entry['diff'])
+        entry['diff'] = PSS(entry['diff'])
```

Comment:
Is this necessary?

Reply:
I think so, yeah. If we don't use `PSS`, we'll end up with difficult-to-read `diff` properties.

Reply:
yeah yeah. but the `diff` already is `PSS` (if we assume only our scripts are making these files). Anyways, doesn't matter that much.

Reply:
I just dropped `PSS`, and you're right: the original formatting is preserved.